tests/allocator-test.cc

```diff
@@ -643,7 +641,8 @@ namespace
 		size_t allocSize       = 128;
 		auto   mallocQuotaLeft = heap_quota_remaining(MALLOC_CAPABILITY);
 		CHERI::Capability alloc{
-		  heap_allocate(&noWait, MALLOC_CAPABILITY, allocSize)};
+		  heap_allocate_cpp(&noWait, MALLOC_CAPABILITY, allocSize)
+		    .as_raw_capability()};
```

Comment:
Stylistic, but I think this could be an assignment, `alloc = heap_allocate_cpp(...).as_raw_capability();` rather than invoking the copy constructor?